docs/proposals/message-proto-design.md

```diff
@@ -0,0 +1,297 @@
+---
+title: change model message to  protobuf message 
+authors:
+    - "@kadisi"
+    - "@rohitsardesai83"
+approvers:
+  - "@CindyXing"
+  - "@sids-b"
+  - "@qizha"
+  - "kevin-wangzefeng"
+creation-date: 2019-07-11
+last-updated: 2019-07-24
+status: alpha
+---
+
+# change model message to protobuf message 
+
+* [change model message to protobuf message](#change-model-message-to-protobuf-message)
+   * [Abstract](#abstract)
+   * [Need to do](#need-to-do)
+   * [How to do](#how-to-do)
+   * [Change the model message to protobuf message  directly](#change-the-model-message-to-protobuf-message--directly)
+   * [Solve the filed index problem of Resource](#solve-the-filed-index-problem-of-resource)
+   * [Consider compatibility](#consider-compatibility)
+* [How to use in kubeedge project](#how-to-use-in-kubeedge-project)
+* [Which projects need to be modified](#which-projects-need-to-be-modified)
+
+## Abstract 
+
+currently, In beehive framework, model message is defined and serialized using json format. the performance is slower than protobuf.
+
+and for parsing the message resource, each field index is hardcoded. such as (https://github.com/kubeedge/beehive/blob/master/pkg/common/util/parse_resource.go)
+
+in kubeedge project , we can see mush `Split(string, "/")` function which to parse the resource path string.  these are hard code.   when we deal will the message resource after we receive message  . we need to know the format of the assembly In advance。 then use `Split()` function to get the field we want.    it`s not easy to expand. It's not so extensible. so we need a more extended form.
```

Comment:
kubeedge -> KubeEdge, applies to all such occurences.

Reply:
Ok